src/pages/module-builder/Header.tsx

```diff
@@ -85,7 +86,12 @@ const Header = ({
                         onClick={() => {
                             handleSaveModule();
                         }}
-                        isDisabled={state.title === ""}
+                        isDisabled={
+                            state.title === "" ||
+                            (state.lastSavedState &&
+                                state.lastSavedState.title === state.title &&
+                                state.lastSavedState.slides === state.slides)
```

Comment:
since it is only checking for shallow equality here, i don't think the comparison for `state.lastSavedState.slides` works as intended. we can keep the same functionality by just changing this lastSavedState to a boolean that just tracks whether or not an update section action has been called. additionally, we don't need to save an extra copy of the state (which can get large)